assembl/models/idea.py

```diff
@@ -999,11 +999,30 @@ class RootIdea(Idea):
 
     @property
     def num_posts(self):
-        """ In the root idea, this is the count of all mesages in the system """
+        """ In the root idea, num_posts is the count of all non-deleted mesages in the discussion """
+        from .post import Post
+        result = self.db.query(Post).filter(
+            Post.discussion_id == self.discussion_id,
+            Post.hidden==False,
+            Post.tombstone_condition()
+        ).count()
+        return int(result)
+
+    @property
+    def num_read_posts(self):
+        """ In the root idea, num_posts is the count of all non-deleted read mesages in the discussion """
         from .post import Post
```

Comment:
a) Import in the middle of a file does not take longer in Python. It is executed once. I have seen this style in the wild.
b) not familiar with moving imports at the end, worth trying, could be cleaner.